csrc/scheduler/hopper_multi_matmul.cpp

```diff
@@ -499,28 +502,30 @@ void HopperMultipleMatmulScheduler::scheduleEpilogue() {
     for (Val* dv : fusion_->outputs()) {
       auto* d = dv->as<TensorView>();
       NVF_ERROR(d->definition() && d->definition()->isA<LoadStoreOp>());
-      auto* dc = d->definition()->input(0)->as<TensorView>();
 
-      // Block Schedule and Parallelize
-      blockTileTensors({dc, d});
-      parallelizeBlocks({dc, d});
+      blockTileTensors({d});
+      parallelizeBlocks({d});
+      d->split(-2, getM(params_->mma_macro));
+      d->split(-1, getN(params_->mma_macro));
+      // [..., Mo, No, Mio, Mii, Nio, Nii]
+      // -> [..., Mo, No, Mio, Nio, Mii, Nii]
+      d->reorder({{-3, -2}});
+      d->merge(-4);
+      auto s = mma_utils::MmaSwizzler::scheduleMmaOutputAllocation(
+          d->getLoopDomain());
+      d->setLoopDomain(s.as<IterDomain*>());
+      d->axis(-5)->parallelize(ParallelType::TIDy);
+
+      scheduler_utils::BoundedDirectionalTransformPropagator::backward(
+          d,
+          -1,
+          mma_results_,
+          scheduler_utils::BoundedDirectionalTransformPropagator::Options()
+              .propagateParallelType());
 
-      // Apply mma common transformation
-      for (auto tv : {dc, d}) {
-        // [..., Mo, No, Mi, Ni]
-        tv->split(-2, getM(params_->mma_macro));
-        tv->split(-1, getN(params_->mma_macro));
-        // [..., Mo, No, Mio, Mii, Nio, Nii]
-        // -> [..., Mo, No, Mio, Nio, Mii, Nii]
-        tv->reorder({{-3, -2}});
-        tv->merge(-4);
-        auto s = mma_utils::MmaSwizzler::scheduleMmaOutputAllocation(
-            tv->getLoopDomain());
-        tv->setLoopDomain(s.as<IterDomain*>());
-        tv->axis(-5)->parallelize(ParallelType::TIDy);
-      }
       d->axis(-1)->parallelize(ParallelType::Vectorize);
     }
+    scheduleFusionInputsForEpilogue();
```

Comment:
In this path, since we are not using TMA we should respect the `params_->supported_vec_size.epilogue` parameter to set the vectorization width. See `scheduleOutputTensor` in this file, which we should either update and use or remove.

In the TMA path (the else branch) we might not be able to use `scheduleFusionInputsForEpilogue` as is because it is currently propagating the schedule back from the output `d`, which is scheduled for a TMA store not a vectorized load. We could potentially just do that propagation then check the innermost dims and merge/split/unroll as needed in order to hit the target vectorization for the epilogue.